### DISCUSSION TITLE
Directly communicating with a mote

### DISCUSSION THREAD

**Discussion by rahav** [contiki-ng/contiki-ng#2832] [Q&A]
I would like to send short messages to motes, not through a border router. Is there an established/recommended method for communicating directly with mote from a Linux computer (e.g. Raspberry Pi) the mote is attached to?

Is there Contiki API that allow that?

**Comment by alexrayne** [comment]
Boarder router alredy have comm by  ip under slip. You can look it - how
this bridge implemented.

ср, 3 янв. 2024 г., 23:25 Rahav Dor ***@***.***>:

**Reply by rahav** [reply]
Thanks Alex. I am tryin to understand the approach your are suggesting. Do you mean using parts of the boarder router, on each RPi/Mote that needs this communication?

**Comment by alexrayne** [comment]
If you talk about ip protocol communicatation, naive way - just how it
alredy implemented. I guess br use uart-slip driver bridge to network
driver.

If you talk about some other communication, there is another talk. For
example conriki have command line processing over uart. Maybe it more
intrest for you.

чт, 4 янв. 2024 г., 20:11 Rahav Dor ***@***.***>:

**Reply by rahav** [reply]
Thanks again @alexrayne and I would like to try both ways and see which one is more stable and flexible. The general idea I have in mind is using Python running on the RPi OS, which then directly communicates with the attached mote.

The reason I want that is to prevent control commands from traversing over the low-power network, from the root node to the node that needs to receive the command (for example, change of configuration command).

Can you please point me to the API you mentioned for command line processing over uart?
